src/js/program/media-controller.js

```diff
@@ -280,7 +284,12 @@ export default class MediaController extends Events {
                 if (provider.removeFromContainer) {
                     provider.removeFromContainer();
                 } else {
-                    container.removeChild(provider.video);
+                    if (__HEADLESS__) {
```

Comment:
Sorry for nitpick - this can be `else if` and `else`

Reply:
I prefer not to chain headless blocks like this. Keeping it in it's own scope makes it clearer what the headless player does vs normal operation.